### ISSUE TITLE
Improve the code injection

### ISSUE THREAD

**Issue by mkaring** [mkaring/ConfuserEx#2]
The code injection is currently very … simple. It can't handle dependencies to other classes within the runtime assembly and it does not check if the injected code is actually properly referencing existing members in the new assembly.

All these problems need to be fixed.

**Comment by bprg** [comment]
Is possible using dnSpy `merger` instead of current injection?

**Comment by mkaring** [comment]
The implementation of the dnSpy `merger` and the inject code of ConfuserEx work very similar. Both utilize the `Importer` that is provided by dnlib. The problem with the ConfuserEx is that the assembly resolver that is required for the `Importer` is not properly implemented. That is what I am going to fix.

All in all it will be very similar to the dnSpy implementation. In both cases dnlib will do the heavy lifting.

**Comment by mkaring** [comment]
I had a look at the merger, but that is not really working in the case required here. The copies the source assembly fully into the target assembly. That is not what is required here. The code injection is only expected to copy a very select number of classes into the target and it is supposed to modify the injected classes (rename, moving them around).

How ever I think something like the merger will still be required, to enable confuser to merge dependency assemblies as part of the obfuscation process.

**Comment by mkaring** [comment]
I am pretty much done with the rework. The new code injection handled all kinds of dependencies nicely. I updated the LZMA implementation to the current unmodified version. So it works nicely even if the required classes and methods are not all in the same class.

The new unit tests are working and I tested the changes by obfuscating ILSpy. I'll do a few more tests and obfuscate some more projects and then this is done.
